src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java

```diff
@@ -137,8 +139,10 @@ public void run(@NotNull ProgressIndicator progressIndicator) {
 
                 // Finished
                 codeSyncProgressIndicator.setMileStone(InitRepoMilestones.END);
+                pluginState.syncInProcess = false;
```

Comment:
Do we have setState as well? like we have `StateUtils.getState`

Reply:
How about if API call is failed or some other issue happend before inite-request is finished? @syedgulahmed

Reply:
> Do we have setState as well? like we have `StateUtils.getState`

Yes, populateState() is doing that job!

Reply:
> How about if API call is failed or some other issue happend before inite-request is finished? @syedgulahmed

Yes, we have to put checks in case failure of connecting repo. There are lot of processes going on while connecting repo, we have to go through each process and make sure to put check for each of them. I was testing on prod, only testing I could do was disconnecting internet while connecting repo. I will need help of Saleem on this, probably I have to do a session with him.

Reply:
Got it, thanks

Reply:
@syedgulahmed I do think changes in `src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetup.java` are enough to mark if repo sync is in progress or has completed. 
The changes you made in `src/main/java/org/intellij/sdk/codesync/clients/CodeSyncClient.java` are redundant.

Reply:
Yeah, but it is not working when server is down or there is connection issue. It is making Connect button disable permanently in that scenario.